### ISSUE TITLE
Roadmap - `trl` 0.2

### ISSUE THREAD

**Issue by younesbelkada** [lvwerra/trl#64]
A list of cool things that we can aim for `trl` 0.2! :

## API:

- [x] general decoder models support (OPT, BLOOM, GPT-J, etc): https://github.com/lvwerra/trl/pull/53
- [x] `accelerate` integration for training in mixed precision, DP (multi-GPU), using DeepSpeed: https://github.com/lvwerra/trl/pull/58 
- [x] shared backbones for lighter training: https://github.com/lvwerra/trl/pull/61 
- [x] Once ^ is merged (or inside the PR) - make the reference model optional and create both models inside the trainer if no reference model is passed |  https://github.com/lvwerra/trl/pull/67
- [x] soften requirements : https://github.com/lvwerra/trl/pull/66 
- [x] Make quality & make style: https://github.com/lvwerra/trl/pull/62
- [x] use accelerator logger instead of wandb to remove hard dependency (for users that do not want to use `wandb`) | https://github.com/lvwerra/trl/pull/92 
- [X] Refactor Trainer (push to hub etc. cc @lewtun) https://github.com/lvwerra/trl/pull/68
- [x] general encoder-decoder models support | https://github.com/lvwerra/trl/pull/93
- [x] Support for Adam 8bit (for lighter training)? | https://github.com/lvwerra/trl/pull/78
- [x] clean `step` (sanity checks) | https://github.com/lvwerra/trl/pull/76
- [x] `dataset` attribute should be optional ? | https://github.com/lvwerra/trl/pull/85
- [x] Add learning rate scheduler support | https://github.com/lvwerra/trl/pull/96
- [x] Handle: https://github.com/lvwerra/trl/pull/68#discussion_r1062268630 | https://github.com/lvwerra/trl/pull/86

## Documentation

- [x] doc build: https://github.com/lvwerra/trl/pull/59 
- [x] document API (classes, functions) | https://github.com/lvwerra/trl/pull/91
- [ ] document tips and good practices when doing RLHF (hyper-parameters etc.) / https://iclr-blog-track.github.io/2022/03/25/ppo-implementation-details/ 
- [x] clean up README (examples don't work anymore)  https://github.com/lvwerra/trl/pull/70

## Improvements

- [ ] add different example than fine-tuning gpt2 on sentiment analysis
- [ ] add an example with a custom reward model (SetFit?)
- [x] convert the notebook `05` https://github.com/lvwerra/trl/pull/80
- [ ] What is the largest model we can train on?
- [ ] What is the coolest dataset we can use to train a model (e.g. anthropic AI dataset: https://huggingface.co/datasets/Anthropic/hh-rlhf ?
- [x] Make all internal imports relative (e.g. `from trl.trainer` -> `from .trainer`) https://github.com/lvwerra/trl/pull/70
- [x] Clean up `setup.py` and remove `settings.ini` (legacy from `nbdev`) https://github.com/lvwerra/trl/pull/70
- [x] Create & add a logo | https://github.com/lvwerra/trl/pull/95

Any suggestion very welcome!

**Comment by LouisCastricato** [comment]
BTW, I can confirm that SetFit does make for a really good zero shot RM. There are some issues with using contrastive models as RMs though. It often requires very careful data cleaning and identifying what kinds of clusters work as RMs is a dark art to the point where we decided that it wasn't worth seriously pursing further after CARP CoOp. Rerank models are much better.

**Comment by TristanThrush** [comment]
I think that the "coolest" dataset we can use to train a model could be https://huggingface.co/datasets/openai/webgpt_comparisons, but it is hard to evaluate this sort of model after we train it. I might start by adding a summarization example, and then some decent ways by which it can be evaluated. Then the webgpt comparisons example

**Comment by AlexWortega** [comment]
https://colab.research.google.com/drive/1hkPBFtMP5xBAjNYMjWH7NqYn118kRLOJ?usp=sharing
I am trying to implement own gpt + trl with QA retrival reward, but i think something is wrong with reward/or generation

**Comment by natolambert** [comment]
@AlexWortega can you open a separate issue / PR for this? Looks interesting, but may get loss in this big 1.0 roadmap thread.

**Comment by lvwerra** [comment]
We ended up calling this release `0.2` (not `1.0`). I am closing the issue and will move the open tasks to a new issue.

**Comment by AlexWortega** [comment]
Hi @lvwerra  i opened PR https://github.com/lvwerra/trl/pull/149  with this feature(?) idea
